tiledb/common/dynamic_memory/test/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+#
+# tiledb/common/dynamic_memory/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2021-2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(unit_test)
+
+commence(unit_test dynamic_memory)
+    this_target_object_libraries(baseline)
+    this_target_sources(main.cc unit_dynamic_memory.cc)
+    # Link to Threads::Threads if library or flag needed to enable threading.
+    # The heap_profiler dependency requires threads. The dependency should be
+    # moved to heap_profiler if/when it is added to its own object library.
+    if (CMAKE_THREAD_LIBS_INIT)
+        target_link_libraries(unit_dynamic_memory PUBLIC Threads::Threads)
```

Comment:
One thing that I'm seeing that's a bit awkward is when we have these odd cases where we have to break out to "normal" CMake functions. The unit name listed on line 29 is just `dynamic_memory` which gets the `unit_` prefix prepended in the new macros, but when we need to refer to it in these extra config bits we have to remember to specify the full `unit_dynamic_memory`.

Its not the end of the world, but I worry slightly that it would be easy to accidentally elide the `unit_` prefix and then end up declaring config options on an object library so that things appear to work but aren't quite right.

Given all of that, I wonder if we shouldn't require the `unit_` prefix on line 29 in the comment macro call (which asserts that all unit test targets have that prefix to avoid this possible confusion? Or alternatively if there's a magic way to have a variable `CURRENT_UNIT_TARGET` or some such that would expand automagically instead.

Regardless, I'll defer to you on whether this can be meaningfully improved.

Reply:
> `CURRENT_UNIT_TARGET`

I'll call it `this_target()`, but it's a very good suggestion.

Reply:
It's still a good suggestion, but I've decided that it's beyond the scope of this PR. Doing it anywhere near right require a new primitives in `environment.cmake` and new specifications for environment-sensitive variables. It's worth doing at some point but is too much work to do now.

Reply:
More than fair. It also occurs to me that while its a bit of a foot gun, its not like we change the build system all that often anyway.

Reply:
I'll merge this as-is, but I'd like to see this change done soon as a follow-up. From past experience I strongly dislike automatic target names.